packages/lisk-p2p/src/peer.ts

```diff
@@ -12,59 +12,167 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-/* tslint:disable:interface-name variable-name */
-import { PeerState } from './p2p_types';
+/* tslint:disable:interface-name */
+import {
+	P2PRequestPacket,
+	P2PMessagePacket,
+	P2PResponsePacket,
+	NodeInfo,
+} from './p2p_types';
 
-// TODO: Use to create outbound socket connection inside peer object.
-// TODO: const socketClusterClient = require('socketcluster-client');
+import socketClusterClient from 'socketcluster-client';
 
 export interface PeerConfig {
 	readonly clock?: Date;
-	readonly height: number;
+	readonly height?: number;
 	readonly id: string;
 	/* tslint:disable:next-line: no-any */
 	readonly inboundSocket?: any; // TODO: Type SCServerSocket
 	readonly ipAddress: string;
 	readonly os?: string;
 	readonly version?: string;
 	readonly wsPort: number;
+	readonly nodeInfo: NodeInfo;
+}
+
+export enum ConnectionState {
+	CONNECTING = 0,
+	CONNECTED = 1,
+	DISCONNECTED = 2,
+}
+
+export interface PeerState {
+	readonly inbound: ConnectionState;
+	readonly outbound: ConnectionState;
+}
+
+interface RawResponseBody {
+	readonly data: unknown;
 }
 
 export class Peer {
 	private readonly _height: number;
 	private readonly _id: string;
-	private readonly _inboundSocket: any;
+	private _inboundSocket: any;
+	private _outboundSocket: any;
 	private readonly _ipAddress: string;
 	private readonly _wsPort: number;
+	private readonly _nodeInfo: NodeInfo;
 
 	public constructor(peerConfig: PeerConfig) {
 		this._id = peerConfig.id;
 		this._ipAddress = peerConfig.ipAddress;
 		this._wsPort = peerConfig.wsPort;
 		this._inboundSocket = peerConfig.inboundSocket;
-		this._height = peerConfig.height;
+		this._nodeInfo = peerConfig.nodeInfo;
+		this._height = peerConfig.height === undefined ? 0 : peerConfig.height;
 	}
-	// TODO: Return BANNED when appropriate.
-	public get state(): PeerState {
-		if (this._inboundSocket.state === this._inboundSocket.OPEN) {
-			return PeerState.CONNECTED;
-		}
 
-		return PeerState.DISCONNECTED;
+	public connect(): void {
+		const nodeInfo = this._nodeInfo;
+		this._outboundSocket = socketClusterClient.create({
+			hostname: this._ipAddress,
+			port: this._wsPort,
+			query: {
```

Comment:
`query: this._nodeInfo`? since they have the same property names? or, maybe its good for clarity?